File: ComplexJava/src/complex/CarComplex.java
package complex;

public class CarComplex extends AbsComplex {
	private double real;
	private double imag;

	public CarComplex(double real, double imag){
		this.real = real;
		this.imag = imag;
	}
	public CarComplex(double radius, Angle azimuth){
		this.real = Conversion.real(radius, azimuth);
		this.imag = Conversion.imag(radius, azimuth);
	}

	public double getReal(){
		return real;
	}
	public double getImag(){
		return imag;
	}
	public double getRadius(){
		return Conversion.radius(this.getReal(), this.getImag());
	}
	public Angle getAzimuth(){
		return Conversion.azimuth(this.getReal(), this.getImag());
	}

	public Complex create(double real, double imag){
		return new CarComplex(real, imag);
	}
	public Complex create(double radius, Angle azimuth){
		return new CarComplex(radius, azimuth);
	}
}

File: ScalaComplex/src/complex/CarComplex.scala
package complex

class CarComplex(val real: Double, val imag: Double) extends AbsComplex {
  def this(radius: Double, azimuth: Angle) {
    this(Conversion.real(radius, azimuth), Conversion.imag(radius, azimuth))
  }

  override val radius: Double = Conversion.radius(this.real, this.imag)
  override val azimuth: Angle = Conversion.azimuth(this.real, this.imag)

  override def create(real: Double, imag: Double): Complex = new CarComplex(real, imag)
  override def create(radius: Double, azimuth: Angle): Complex = new CarComplex(radius, azimuth)
}